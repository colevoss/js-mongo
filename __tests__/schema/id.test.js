const Schema = require('../../src/Schema');
const { ObjectId } = require('mongodb');

describe('Schema.id', () => {
  test('Returns an ObjectId for valid id string or ObjectId', () => {
    const testData = { test: '5acebe0610348ab07dde7672' };
    const oTestData = { test: new ObjectId() };

    expect(Schema.id('test', testData).toHexString()).toBe(testData.test);

    expect(Schema.id('test', oTestData).toHexString()).toBe(
      oTestData.test.toHexString(),
    );
  });

  test('Returns error', () => {
    const testData = { test: 'INVALID ID' };

    const checked = Schema.id('test', testData);

    expect(checked).toBeInstanceOf(TypeError);
    expect(checked).toMatchSnapshot();
  });

  test('__type', () => {
    expect(Schema.id.__type).toBe('id');
  });
});
