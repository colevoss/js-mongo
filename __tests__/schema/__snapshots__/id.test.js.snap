// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.id Returns error 1`] = `[TypeError: Expected test to be an valid ObjectId value but recieved INVALID ID]`;
